### PR TITLE
C# protected property can be assigned in a F# inherit constructor call

### PR DESCRIPTION
**Pull request by edgarfgp**
## Description

Fixes https://github.com/dotnet/fsharp/issues/13299

## Checklist

- [x] Test cases added
- [x] Release notes entry updated

### CODE REVIEW

File: src/Compiler/Checking/TypeRelations.fs [hunk 1]
```diff
@@ -117,16 +117,16 @@ let rec TypeFeasiblySubsumesType ndeep g amap m ty1 canCoerce ty2 =
...
-        (isObjTy g ty1 && (canCoerce = CanCoerce || isRefTy g ty2)) 
-        ||
-        (isAppTy g ty2 &&
-         (canCoerce = CanCoerce || isRefTy g ty2) && 
-         begin match GetSuperTypeOfType g amap m ty2 with 
-         | None -> false
-         | Some ty -> TypeFeasiblySubsumesType (ndeep+1) g amap m ty1 NoCoerce ty
-         end ||
-         ty2 |> GetImmediateInterfacesOfType SkipUnrefInterfaces.Yes g amap m 
-             |> List.exists (TypeFeasiblySubsumesType (ndeep+1) g amap m ty1 NoCoerce))
```
Comment (vzarytovskii) [inline src/Compiler/Checking/TypeRelations.fs:129]: Before all these were executed "lazily", if left side of OR was true. Now they all get executed unconditionally, which is more expensive.

Comment (edgarfgp) [reply src/Compiler/Checking/TypeRelations.fs:129]: Yeah I’m  just trying to debug this. Will put improve/revert it once I know what is going on

File: src/Compiler/Checking/MethodCalls.fs [hunk 2]
```diff
@@ -1237,7 +1237,8 @@ let MethInfoChecks g amap isInstance tyargsOpt objArgs ad m (minfo: MethInfo)  =
                 AccessibleFrom(paths, None) 
         | _ -> ad
 
-    if not (IsTypeAndMethInfoAccessible amap m adOriginal ad minfo) then 
+    let isProtectedSetter = minfo.IsProtectedAccessibility && minfo.LogicalName.StartsWithOrdinal("set_")
+    if not (IsTypeAndMethInfoAccessible amap m adOriginal ad minfo) && not isProtectedSetter then
```
Comment (vzarytovskii) [inline src/Compiler/Checking/MethodCalls.fs:1241]: I wonder if it makes sense to inline the check, so it's only performed if LHS is `true`?

```fsharp
if false && (printfn "side-effect"; true) then printfn "yes";;
```

Comment (edgarfgp) [reply src/Compiler/Checking/MethodCalls.fs]: Sure. Thanks for the review :)

**Review by vzarytovskii — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]
Thanks Edgar, nice fix!

**Comment by edgarfgp** [comment]
🍻

**Pull request merged**